sources/api/pluto/src/main.rs

```diff
@@ -270,11 +277,70 @@ fn pause_container_uri(region: &str) -> Result<String> {
     ))
 }
 
+/// Calculates the amount of memory to reserve for kubeReserved in mebibytes.
+fn get_memory_mebibytes_to_reserve(client: &Client, session_token: &str) -> Result<String> {
+    let max_num_pods = get_max_pods(&client, &session_token)?;
+    let mebibytes_unit = "Mi";
+    let max_num_pods_u32 = max_num_pods.parse::<u32>().context(error::StringParseU32 { max_num_pods })?;
+    let memory_to_reserve = max_num_pods_u32 * 11 + 255;
+
+    Ok(memory_to_reserve.to_string() + mebibytes_unit)
+}
+
+/// Calculates the amount of CPU to reserve for kubeReserved in millicores
+/// from the total number of vCPUs available on the instance.
+/// We are using these CPU ranges from GKE
+/// (https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-architecture#node_allocatable):
+///  6% of the first core
+///  1% of the next core (up to 2 cores)
+///  0.5% of the next 2 cores (up to 4 cores)
+///  0.25% of any cores above 4 cores
+fn get_cpu_millicores_to_reserve() -> Result<String>{
+    let total_cpu_on_instance = num_cpus::get() * 1000;
+    let cpu_ranges = vec![0, 1000, 2000, 4000, total_cpu_on_instance];
+    let cpu_percentage_reserved_for_ranges = vec![600, 100, 50, 25];
+    let mut cpu_to_reserve = 0;
+    let millicores_unit = "m";
+
+    for i in 0..(cpu_percentage_reserved_for_ranges.len() - 1) {
+        let start_range = cpu_ranges[i];
+        let end_range = cpu_ranges[i+1];
+        let percentage_to_reserve_for_range = cpu_percentage_reserved_for_ranges[i];
+
+        cpu_to_reserve += get_resource_to_reserve_in_range(
+            total_cpu_on_instance, start_range, end_range, percentage_to_reserve_for_range
+        )
+    }
+
+    Ok(cpu_to_reserve.to_string() + millicores_unit)
+}
+
+fn get_resource_to_reserve_in_range(
+    total_resource_on_instance: usize, start_range: usize, mut end_range: usize, percentage: usize
+) -> usize{
+    let mut resources_to_reserve = 0;
+    if total_resource_on_instance > start_range {
+        if total_resource_on_instance < end_range {
+            end_range = total_resource_on_instance;
+        };
+        resources_to_reserve = (end_range - start_range) * percentage / 100 / 100;
+        }
+    resources_to_reserve
+}
+
 /// Print usage message.
 fn usage() -> ! {
     let program_name = env::args().next().unwrap_or_else(|| "program".to_string());
     eprintln!(
-        r"Usage: {} [max-pods | cluster-dns-ip | node-ip | pod-infra-container-image]",
+        r"Usage: {}
+        [
+            cpu-reserved |
+            memory-reserved |
+            max-pods |
+            cluster-dns-ip |
+            node-ip |
+            pod-infra-container-image
+        ]",
```

Comment:
Was this something done by cargo fmt? I think if we want to make this multi-line, we can do something like this:

```suggestion
        r"Usage: {} SUBCOMMAND
    Subcommands:
        reserved
        memory-reserved
        max-pods
        cluster-dns-ip
        node-ip
        pod-infra-container-image
```

Reply:
oh no, I made it multi-line. I think one line should be fine as well